pkg/registry/core/service/allocator/bitmap.go

```diff
@@ -70,6 +71,50 @@ func NewAllocationMap(max int, rangeSpec string) *AllocationBitmap {
 		rangeSpec: rangeSpec,
 	}
 	return &a
+
+}
+
+// DynamicRangeOffset estimates the offset used on the range for statically allocation based on
+// the following formula `min(max($min, cidrSize/$step), $max)`, described as ~never less than
+// $min or more than $max, with a graduated step function between them~. The function returns 0
+// if any of the parameters is invalid.
+func DynamicRangeOffset(min, max, step, cidrSize int) int {
+	if cidrSize < min || step > cidrSize || min > max {
+		return 0
+	}
+	offset := cidrSize / step
+	if offset < min {
+		return min
+	}
+	if offset > max {
+		return max
+	}
+	return offset
+}
+
+// NewAllocationReservedMap creates an allocation bitmap using a random scan strategy that,
+// if the maximum value is greater than 64, is subdivided in two blocks.
+// The first block of values will not be used for random value assigned by the AllocateNext()
+// method until the second block of values has been exhausted.
+func NewAllocationReservedMap(max int, rangeSpec string) *AllocationBitmap {
+	// default $min = 16, $max = 256 and $step = 16.
+	offset := DynamicRangeOffset(16, 256, 16, max)
+
+	if offset == 0 {
+		return NewAllocationMap(max, rangeSpec)
+	}
+	a := AllocationBitmap{
+		strategy: randomScanReservedStrategy{
+			rand:     rand.New(rand.NewSource(time.Now().UnixNano())),
```

Comment:
I don't like that this copies the code from `NewAllocationMap()` - why not make `NewAllocationMap()` call this new function with a "reservedSize" param of 0, and let IP layer call it with a non-zero size.

Then the random init and per-field init is only done once

Reply:
I wanted to avoid to create the dependency on the IP layer, the reserved size is calculated here, in L101

> 	offset := dynamicRangeOffset(16, 256, 16, max)

but now that you mention, I did create that dependency in a later commit and sounds wrong that the bitmap is the one calculating the offset, I will refactor as you say